pywr-core/src/recorders/aggregator.rs

```diff
@@ -47,7 +48,7 @@ impl AggregationFrequency {
         let mut sub_values = Vec::new();
 
         let mut current_date = value.start;
-        let end_date = value.start + value.duration;
+        let end_date = value.start + *value.duration.time_delta();
```

Comment:
One alternative  here would be to implement `Add` for `PywrDuration` with `rhs = NaiveDateTime`